sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

```diff
@@ -61,13 +61,16 @@ private CosmosClientBuilderHelper() {}
 
         public static void setCosmosClientBuilderAccessor(final CosmosClientBuilderAccessor newAccessor) {
             if (!accessor.compareAndSet(null, newAccessor)) {
-                logger.warn("CosmosClientBuilderAccessor already initialized!");
+                logger.debug("CosmosClientBuilderAccessor already initialized!");
+            } else {
+                logger.info("Setting CosmosClientBuilderAccessor...");
+                cosmosClientBuilderClassLoaded.set(true);
```

Comment:
Wouldn't setting the accessor have the same race condition issue that was with getter?

Reply:
Yes - there is a race. Multiple threads might call setxxxAccessor concurrently - only one thread wil "win" (enter the else statement) - this "winner" will have set the accessor and also the the xxxClassLoaded flag. So threads calling into getxxxAccessor will only not call into the xxx.initialize (with calls setxxxAcccessor) after we are sure the accessor has been set. So, yes - there is a race - and yes - it is thread-safe now.